server/src/service/initialize/initialize-service.js

```diff
@@ -80,85 +82,142 @@ async function accountFollowingService(api, id, db) {
+        const curFollowersIds = new Set(curFollowers.keys());
+        const curNeighbors = [...curFollowing, ...curFollowers];
 
-            // Process followers and followings for the account
-            for (const followerId in followers) {
-                const follower = followers[followerId];
-                if (!relations[follower.id]) {
-                    relations[follower.id] = new Set();
-                }
-                relations[follower.id].add(account.id);
-                accountInfoMap.set(follower.id, follower);
+        const newNeighbors = new Map(); // Filter out seen neighbors which are already in the heap
+        for (const [nodeId, nodeInfo] of curNeighbors) {
+            if (!seenNodes.has(nodeId)) {
+                newNeighbors.set(nodeId, nodeInfo);
+                seenNodes.add(nodeId);
+            } else if (accountInfoMap.has(nodeId)) {
+                // update depth if shorter path is found
+                curNodeInfo['depth'] = Math.min(
+                    curNodeInfo['depth'],
+                    accountInfoMap.get(nodeId)['depth'] + 1
+                );
             }
+        }
+
+        // Update current node info
+        curNodeInfo['following'] = curFollowingIds;
+        if ('followingSource' in curNodeInfo) {
+            curNodeInfo['following'].add(curNodeInfo['followingSource']);
+        }
+        accountInfoMap.set(curNodeInfo['id'], curNodeInfo);
 
-            for (const followingId in followings) {
-                const following = followings[followingId];
-                if (!relations[account.id]) {
-                    relations[account.id] = new Set();
-                }
-                relations[account.id].add(following.id);
-                accountInfoMap.set(following.id, following);
+        for (const [nodeId, nodeInfo] of newNeighbors) {
+            nodeInfo['depth'] = curNodeInfo['depth'] + 1;
+            nodeInfo['priority'] = calcPriority(nodeInfo);
+
+            // Need to add this connection directly bc. api only returns a subset of following,
+            // which may not contain the account which led to that account and would lead to a
+            // disconnected graph.
+            if (curFollowersIds.has(nodeId)) {
+                nodeInfo['followingSource'] = curNodeInfo['id'];
             }
+
+            nodeHeap.push(nodeInfo);
         }
     }
 
-    // Fetch and process first-level followers and followings
-    const firstLevelFollowers = await accountFollowersService(api, id, db);
-    const firstLevelFollowings = await accountFollowingService(api, id, db);
-    await processAccounts(firstLevelFollowers, true);
-    await processAccounts(firstLevelFollowings, false);
+    // Update node follower info to only contain nodes in accountInfoMap
+    const selectedNodes = new Set(accountInfoMap.keys());
+    for (const nodeInfo of accountInfoMap.values()) {
+        const removeUnusedFollower = (nodeId) =>
```

Comment:
again this is probably just personal preference :) I can also see you way doesn't look like a nested loop, but I will include this anyways...

```
// remove unused followers
nodeInfo['following'].forEach((nodeId) => {
     if (!selectedNodes.has(nodeId)) {
             nodeInfo['following'].delete(nodeId)
     }
});
```

Reply:
good call, updated